### ISSUE TITLE
FLEBasis2D rotations incorrect under `match_fb=True`

### ISSUE THREAD

**Issue by garrettwrong** [ComputationalCryoEM/ASPIRE-Python#907]
One of the first tasks for reconciling our coefs/basis is ensuring we have `rotation` available across all the steerable 2d basis classes.  Testing in a notebook I found the rotations are incorrect when using `FLEBasis2D` in the `match_fb` mode. 

I took `tests/test_FLEbasis2D.py`  and altered all instances of `match_fb=False` to `match_fb=True`.  This showed that we are actually failing tests for at least what is below.

It may be possible to move forward without addressing this by using the original (match_fb=False) mode, but it is definitely something that needs to be fixed up.  I'm taking a look now to see if it is another small bug, but hasn't popped out yet.

<img width="1135" alt="Screen Shot 2023-04-07 at 2 14 03 PM" src="https://user-images.githubusercontent.com/47759732/230657516-95dc42e5-9b62-45c5-a55b-8f0ba2ed35b9.png">

```
=================================== short test summary info ====================================
FAILED tests/test_FLEbasis2D_match_fb.py::TestFLEBasis2D::testFastVDense[32-0.0001] - assert 1.335244139991674 < 0.0001
FAILED tests/test_FLEbasis2D_match_fb.py::TestFLEBasis2D::testFastVDense[32-1e-07] - assert 1.3352472702407219 < 1e-07
FAILED tests/test_FLEbasis2D_match_fb.py::TestFLEBasis2D::testFastVDense[32-1e-10] - assert 1.3352472712879315 < 1e-10
FAILED tests/test_FLEbasis2D_match_fb.py::TestFLEBasis2D::testFastVDense[32-1e-14] - assert 1.3352472712873953 < 1e-14
FAILED tests/test_FLEbasis2D_match_fb.py::TestFLEBasis2D::testFastVDense[33-0.0001] - assert 1.3298222477525323 < 0.0001
FAILED tests/test_FLEbasis2D_match_fb.py::TestFLEBasis2D::testFastVDense[33-1e-07] - assert 1.3298229767551073 < 1e-07
FAILED tests/test_FLEbasis2D_match_fb.py::TestFLEBasis2D::testFastVDense[33-1e-10] - assert 1.3298229752582897 < 1e-10
FAILED tests/test_FLEbasis2D_match_fb.py::TestFLEBasis2D::testFastVDense[33-1e-14] - assert 1.3298229752613686 < 1e-14
FAILED tests/test_FLEbasis2D_match_fb.py::TestFLEBasis2D::testFastVDense[8-0.0001] - assert 1.2013107587239196 < 0.0001
FAILED tests/test_FLEbasis2D_match_fb.py::TestFLEBasis2D::testEvaluateExpand[32-0.0001] - RuntimeError: Unable to converge!
FAILED tests/test_FLEbasis2D_match_fb.py::TestFLEBasis2D::testEvaluateExpand[32-1e-07] - RuntimeError: Unable to converge!
FAILED tests/test_FLEbasis2D_match_fb.py::TestFLEBasis2D::testEvaluateExpand[32-1e-10] - RuntimeError: Unable to converge!
FAILED tests/test_FLEbasis2D_match_fb.py::TestFLEBasis2D::testEvaluateExpand[32-1e-14] - RuntimeError: Unable to converge!
FAILED tests/test_FLEbasis2D_match_fb.py::TestFLEBasis2D::testEvaluateExpand[33-0.0001] - RuntimeError: Unable to converge!
FAILED tests/test_FLEbasis2D_match_fb.py::TestFLEBasis2D::testEvaluateExpand[33-1e-07] - RuntimeError: Unable to converge!
FAILED tests/test_FLEbasis2D_match_fb.py::TestFLEBasis2D::testEvaluateExpand[33-1e-10] - RuntimeError: Unable to converge!
FAILED tests/test_FLEbasis2D_match_fb.py::TestFLEBasis2D::testEvaluateExpand[33-1e-14] - RuntimeError: Unable to converge!
FAILED tests/test_FLEbasis2D_match_fb.py::TestFLEBasis2D::testEvaluateExpand[8-0.0001] - RuntimeError: Unable to converge!
FAILED tests/test_FLEbasis2D_match_fb.py::testMatchFBEvaluate[32-0.0001] - assert False
FAILED tests/test_FLEbasis2D_match_fb.py::testMatchFBEvaluate[32-1e-07] - assert False
FAILED tests/test_FLEbasis2D_match_fb.py::testMatchFBEvaluate[32-1e-10] - assert False
FAILED tests/test_FLEbasis2D_match_fb.py::testMatchFBEvaluate[32-1e-14] - assert False
FAILED tests/test_FLEbasis2D_match_fb.py::testMatchFBEvaluate[33-0.0001] - assert False
FAILED tests/test_FLEbasis2D_match_fb.py::testMatchFBEvaluate[33-1e-07] - assert False
FAILED tests/test_FLEbasis2D_match_fb.py::testMatchFBEvaluate[33-1e-10] - assert False
FAILED tests/test_FLEbasis2D_match_fb.py::testMatchFBEvaluate[33-1e-14] - assert False
FAILED tests/test_FLEbasis2D_match_fb.py::testLowPass - assert 9693 == 10014
FAILED tests/test_FLEbasis2D_match_fb.py::testRotate - assert False
FAILED tests/test_FLEbasis2D_match_fb.py::testRadialConvolution - assert False
```

**Comment by garrettwrong** [comment]
In #916 , I've gotten a proof of concept implementation allowing FLE to be used in our existing cov2d implementation.  Presently I'm running in `match_fb=True` mode and the results have similar looking bug.  I suspect it would be the same issue.

![match_fb_fle_cov2d](https://user-images.githubusercontent.com/47759732/233992393-a0550716-8786-41f0-9599-10409a7c0b09.png)

**Comment by garrettwrong** [comment]
Noting that for rotations to work required i) using the fb convention of 1 for sign of `ell==0`, and not reordering the indices or flipping any signs.  I think this makes sense since the rotation code using the index lookup.  This also resolved the above cov2d diff using FLE as well. (again it uses index lookups directly).

I think the looming issue is how to unwind the reordering in practice, places adjusted for match_fb, and make it so we're always consistent.  I'm not sure there is any reason we need to keep the original author ordering once the basis is totally constructed, which might remove the modality outside of the basis function computation.

**Comment by janden** [comment]
>  I'm not sure there is any reason we need to keep the original author ordering once the basis is totally constructed

Agreed. As discussed last week, the only potential use case is for porting code that uses the original ordering. That seems to be an empty set, so we should be good.

**Comment by garrettwrong** [comment]
Should be resolved by #936.
